core-processor/src/ext.rs

```diff
@@ -471,6 +473,15 @@ impl EnvExt for Ext {
         Ok(())
     }
 
+    fn exit_code(&mut self) -> Result<Option<ExitCode>, Self::Error> {
+        self.charge_gas_runtime(RuntimeCosts::ExitCode)?;
+        if let Some(details) = self.context.message_context.current().reply() {
+            Ok(Some(details.into_exit_code()))
+        } else {
+            Ok(None)
+        }
```

Comment:
Can it be rewritten this way? The same is for `reply_to`.
```suggestion
        Ok(self.context.message_context.current().reply().map(|d| d.into_exit_code()))
```